### ISSUE TITLE
Feature Request: Allow for hiding individual aliases

### ISSUE THREAD

**Issue by Keboo** [dotnet/command-line-api#1859]
My current process for renaming aliases looks like the following:
1. Create a new symbol (lets take an Option for example) with the new aliases I want
2. Mark the previous symbol `IsHidden=true`
3. When executing a command, I am not forced to pull both values and determine which gets precedence despite this really only being a single thing. This also makes doing things like validation more difficult.

Alternatively, I could always modify the help build to hide to aliases that I don't want displayed. However, at present, this would require more code, than the above solution.

What I would like is the ability to mark individual aliases as hidden so that I only have a single symbol to manage.

**Comment by jonsequitur** [comment]
> Should the hidden aliases be entirely hidden from completions, typo corrections, and help; or should they be available from completions if the user has already typed part of the hidden alias and there is no other completion? I think I'd prefer hiding them entirely.

Agreed. So far, `IsHidden` means it is hidden everywhere.

**Comment by Keboo** [comment]
Yes I would expect them to be hidden everywhere.

**Comment by mitchcapper** [comment]
Yes given #133 and the seemingly lack of support for considering any case insensitivity say --Start vs --start we add aliases but this clutters up help and tab completion significantly.   If we can hide those options that would be great.
